### ISSUE TITLE
序列值域问题

### ISSUE THREAD

**Issue by LZBNCC** [KimMeen/Time-LLM#131]
您好！我问一下，能对包含负数的时间序列进行预测吗？目前进行了一些实验，貌似预测出来均为正。因此想问一问该如何对包含负数的时间序列进行预测

**Comment by kwuking** [comment]
你好，建议使用norm进行归一化操作处理，确保数据完成规划后进行操作，预测完成后进行inverse恢复原始数据的scale即可。

**Comment by LZBNCC** [comment]
> 你好，建议使用norm进行归一化操作处理，确保数据完成规划后进行操作，预测完成后进行inverse恢复原始数据的scale即可。
我用的是neuralforcast里的模型，是一样的吗？我大致是这么做的，有什么问题吗？图片中的表格是部分结果
self.scaler = StandardScaler()
self.train["y"] = self.scaler.fit_transform(self.train[["y"]])
self.test["y"] = self.scaler.transform(self.test[["y"]])
timellm = TimeLLM(
            h=horizon,
            input_size=horizon * input_size,
            llm=llm,
            llm_config=llm_config,
            llm_tokenizer=llm_tokenizer,
            prompt_prefix=prompt_prefix,
            max_steps=max_steps,
            batch_size=batch_size,
            windows_batch_size=24,
            val_check_steps=val_check_steps,
            loss=loss,
            llm_num_hidden_layers=llm_num_hidden_layers,
            dropout=dropout,
            n_heads=n_heads
        )

        try:
            print(f"正在训练模型: {self.model}")

            nf = NeuralForecast(
                models=[timellm],
                freq='5min'
            )

            nf.fit(df=self.train, val_size=self.val_size, verbose=True)
            forecasts = nf.predict(df=self.test, verbose=True)

            forecasts[self.model] = self.scaler.inverse_transform(forecasts[self.model].values.reshape(-1, 1)).flatten()
            self.test["y"] = self.scaler.inverse_transform(self.test[["y"]])
            
![image](https://github.com/user-attachments/assets/8cb4bd54-ae42-46ed-bb34-d196a6c0b25c)

**Comment by kwuking** [comment]
你好，你是直接使用neuralforecast框架出现预测值和真实值差异较大嘛，训练是否拟合完成呢？

**Comment by LZBNCC** [comment]
> 你好，你是直接使用neuralforecast框架出现预测值和真实值差异较大嘛，训练是否拟合完成呢？

对 主要他最后结果全都是正的，一个负的都没有，并且预测值差异不大
